### ISSUE TITLE
Integer nullable not working

### ISSUE THREAD

**Issue by artel1992** [collerek/ormar#348]
**Describe the bug**
When I create a model with a numeric type id in migrations, this field has a null = true property. but if I specify the type of uuid this property is equal to false.
**To Reproduce**
Model with id as int:
```python 
class Meeting(BaseModel):
    class Meta(MainMeta):
        tablename = 'meetings'
        constraints = [ormar.UniqueColumns("time", "week_day", "teacher_id")]

    id = ormar.Integer(primary_key=True, nullable=False)
    week_day: int = ormar.Integer(minimum=1, maximum=7, choices=[1, 2, 3, 4, 5, 6, 7], nullable=False)
    time: datetime.time = ormar.Time()
    start: datetime.date = ormar.Date()
    end: datetime.date = ormar.Date()
    group: Union[Group, Dict] = ormar.ForeignKey(Group, related_name='meetings', name="group_id", ondelete="CASCADE",
                                                 nullable=False)
    teacher: User = ormar.ForeignKey(User, related_name='meetings', name="teacher_id", ondelete="CASCADE",
                                     nullable=False)
```

migrattions with id as int:
```python
op.create_table('meetings',
    sa.Column('id', sa.Integer(), nullable=True),
    sa.Column('week_day', sa.Integer(), nullable=False),
    sa.Column('time', sa.Time(), nullable=False),
    sa.Column('start', sa.Date(), nullable=False),
    sa.Column('end', sa.Date(), nullable=False),
    sa.Column('group_id', sa.Integer(), nullable=False),
    sa.Column('teacher_id', sa.CHAR(36), nullable=False),
    sa.ForeignKeyConstraint(['group_id'], ['groups.id'], name='fk_meetings_groups_id_group', ondelete='CASCADE'),
    sa.ForeignKeyConstraint(['teacher_id'], ['users.id'], name='fk_meetings_users_id_teacher', ondelete='CASCADE'),
    sa.PrimaryKeyConstraint('id'),
    sa.UniqueConstraint('time', 'week_day', 'teacher_id', name='uc_meetings_time_week_day_teacher_id')
    )
```

Model with id as uuid:
```python
class Meeting(BaseModel):
    class Meta(MainMeta):
        tablename = 'meetings'
        constraints = [ormar.UniqueColumns("time", "week_day", "teacher_id")]

    id = ormar.UUID(primary_key=True, nullable=False)
    week_day: int = ormar.Integer(minimum=1, maximum=7, choices=[1, 2, 3, 4, 5, 6, 7], nullable=False)
    time: datetime.time = ormar.Time()
    start: datetime.date = ormar.Date()
    end: datetime.date = ormar.Date()
    group: Union[Group, Dict] = ormar.ForeignKey(Group, related_name='meetings', name="group_id", ondelete="CASCADE",
                                                 nullable=False)
    teacher: User = ormar.ForeignKey(User, related_name='meetings', name="teacher_id", ondelete="CASCADE",
                                     nullable=False)
```
migrations with id as uuid:
```python
 op.create_table('meetings',
    sa.Column('id', sa.CHAR(32), nullable=False),
    sa.Column('week_day', sa.Integer(), nullable=False),
    sa.Column('time', sa.Time(), nullable=False),
    sa.Column('start', sa.Date(), nullable=False),
    sa.Column('end', sa.Date(), nullable=False),
    sa.Column('group_id', sa.Integer(), nullable=False),
    sa.Column('teacher_id', sa.CHAR(36), nullable=False),
    sa.ForeignKeyConstraint(['group_id'], ['groups.id'], name='fk_meetings_groups_id_group', ondelete='CASCADE'),
    sa.ForeignKeyConstraint(['teacher_id'], ['users.id'], name='fk_meetings_users_id_teacher', ondelete='CASCADE'),
    sa.PrimaryKeyConstraint('id'),
    sa.UniqueConstraint('time', 'week_day', 'teacher_id', name='uc_meetings_time_week_day_teacher_id')
    )
```

migrations are successful, but in the base id has the property notnull = true. and when migrating again, Alembik tries to change this property to false...
```
INFO  [alembic.autogenerate.compare] Detected NULL on column 'meetings.id'
```
second migration:
```python
 op.alter_column('meetings', 'id',
               existing_type=sa.INTEGER(),
               nullable=True,
               autoincrement=True,
               existing_server_default=sa.text("nextval('meetings_id_seq'::regclass)"))
```
and I have error:
```python

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\users\user\appdata\local\programs\python\python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\user\appdata\local\programs\python\python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\User\.virtualenvs\borkovSchool-SnKBJgJY\Scripts\alembic.exe\__main__.py", line 7, in <module>
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\config.py", line 588, in main
    CommandLine(prog=prog).main(argv=argv)
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\config.py", line 582, in main
    self.run_cmd(cfg, options)
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\config.py", line 559, in run_cmd
    fn(
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\command.py", line 320, in upgrade
    script.run_env()
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\script\base.py", line 563, in run_env
    util.load_python_file(self.dir, "env.py")
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\util\pyfiles.py", line 92, in load_python_file
    module = load_module_py(module_id, path)
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\util\pyfiles.py", line 108, in load_module_py
    spec.loader.exec_module(module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "alembic\env.py", line 84, in <module>
    run_migrations_online()
  File "alembic\env.py", line 78, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\runtime\environment.py", line 851, in run_migrations
    self.get_context().run_migrations(**kw)
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\runtime\migration.py", line 612, in run_migrations
    step.migration_fn(**kw)
  File "D:\Projects\Programming\borkovSchool\alembic\versions\96e30f57bf1a_second.py", line 21, in upgrade
    op.alter_column('addresses', 'id',
  File "<string>", line 8, in alter_column
  File "<string>", line 3, in alter_column
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\operations\ops.py", line 1880, in alter_column
    return operations.invoke(alt)
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\operations\base.py", line 387, in invoke
    return fn(self, operation)
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\operations\toimpl.py", line 50, in alter_column
    operations.impl.alter_column(
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\ddl\postgresql.py", line 173, in alter_column
    super(PostgresqlImpl, self).alter_column(
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\ddl\impl.py", line 231, in alter_column
    self._exec(
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\alembic\ddl\impl.py", line 197, in _exec
    return conn.execute(construct, multiparams)
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\sqlalchemy\engine\base.py", line 1263, in execute
    return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\sqlalchemy\sql\ddl.py", line 77, in _execute_on_connection
    return connection._execute_ddl(
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\sqlalchemy\engine\base.py", line 1353, in _execute_ddl
    ret = self._execute_context(
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\sqlalchemy\engine\base.py", line 1814, in _execute_context
    self._handle_dbapi_exception(
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\sqlalchemy\engine\base.py", line 1995, in _handle_dbapi_exception
    util.raise_(
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\sqlalchemy\util\compat.py", line 207, in raise_
    raise exception
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\sqlalchemy\engine\base.py", line 1771, in _execute_context
    self.dialect.do_execute(
  File "c:\users\user\.virtualenvs\borkovschool-snkbjgjy\lib\site-packages\sqlalchemy\engine\default.py", line 717, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.InvalidTableDefinition) ОШИБКА:  столбец "id" входит в первичный ключ

[SQL: ALTER TABLE addresses ALTER COLUMN id DROP NOT NULL]
```
**Expected behavior**
Id is expected to be nullable = false on migrations
**Versions**
 - Database backend used `postgress`
 - Python version `python 3.9.6`
 - `ormar` version `0.10.19`
 - `pydantic` version `1.8.2`
 - `fastapi` version `0.68.1`

**Comment by artel1992** [comment]
I cannot understand Why if the field is the primary key then it should be nullable?
https://github.com/collerek/ormar/blob/14b170097c57f479f042bb25480111acd4894175/ormar/fields/model_fields.py#L90-L92

**Comment by artel1992** [comment]
I fixed it with sql_nullable = False

**Comment by DontPanicO** [comment]
Is this the expected behaviour? should we use sql_nullable = False, or it has to be considered a bug and it's going to be resolved in the future?

**Comment by collerek** [comment]
I will change it. It's a bug. It comes from initial setting of pydantic and database nullable. Setting a primary key makes it not null anyway but it confuses the following migrations.

**Comment by collerek** [comment]
Fixed in 0.10.20

**Comment by artel1992** [comment]
> Fixed in 0.10.20
Thanks
